### ISSUE TITLE
AVT 1 - Code Snippet -> Inline has a DAP violation 

### ISSUE THREAD

**Issue by snidersd** [carbon-design-system/carbon#4529]
## Environment
macOS Mojave version 10.14.6
Chrome Version Version 77.0.3865.90
Carbon v10 - React
DAP IBM Accessibility 1.8.0.0 Sept 2019 Ruleset

## Detailed Description
Go to the React Component 
Run DAP and the following violations are present (see screenshot):
<img width="914" alt="Screen Shot 2019-11-01 at 12 20 22 PM" src="https://user-images.githubusercontent.com/21676914/68042886-02991880-fca2-11e9-910a-943cb256dd6f.png">

Note:  "node -v" which is the button text does not match the aria-label="copyable code snippet".

**Comment by joshblack** [comment]
@snidersd what would the expected resolution be in this case? Seems like we'd want to communicate that the button action is to copy, but having it announce `copy node -v` may be unintended as it would announce the entire code snippet.

**Comment by dakahn** [comment]
Seems like DAP is incorrectly picking up our code text as it's visible label. When really, an inline code snippet has no visible label -- only a programmatic one excluding it from 2.5.3
https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html

**Comment by joshblack** [comment]
Waiting for confirmation on approach from IBMa

**Comment by snidersd** [comment]
@joshblack @dakahn Confirmed that this was discussed with IBMa and the DAP violation is valid, since the code snippet is still considered a visible label.

**Comment by dakahn** [comment]
The internally proposed solution seems to be include the code to be copied in the programmatic label like `aria-label="copy code snippet <div>...`

This doesn't seem like it will scale very well considering there are no technical limits to our inline code snippet. Making the copy button distinct from the code it's acting on is probably the more usable move here I would think.

**Comment by elizabethsjudd** [comment]
@dakahn @snidersd  why not use remove the aria attributes and use the the assistive technology class on a span within the button?  I've personally found it troublesome to use aria-label on elements like buttons and links.

**Comment by dakahn** [comment]
@elizabethsjudd Oh, cool -- thanks for the suggestions we'll keep that in mind. We actually aren't looking to implement the suggested changes as they were proposed. Sometimes a code snippet can be quite long and having that read as a label isn't the experience we want here -- or at least we think it's not.

One idea we're kicking around is having a distinct (copy) button somewhere adjacent to the code. Would make more sense maybe than having the code be a button.

**Comment by jnm2377** [comment]
<img width="1448" alt="Screen Shot 2021-03-17 at 5 36 41 PM" src="https://user-images.githubusercontent.com/32556167/111547752-a2483100-8747-11eb-93ef-690c3618d491.png">
Testing latest carbon react, and I'm not seeing any a11y violations come up. I'm closing this issue for now, but we can re-open if we can reproduce later.
